adapters/silverpush/silverpush.go

```diff
@@ -0,0 +1,333 @@
+
+	return &bannerCopy, nil
+}
+
+func setPublisherId(req *openrtb2.BidRequest, imp *openrtb2.Imp, impExt *openrtb_ext.ImpExtSilverpush) error {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+
+	if err := json.Unmarshal(bidderExt.Bidder, impExt); err != nil {
+		return &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	if impExt.PublisherId == "" {
+		return &errortypes.BadInput{Message: "Missing publisherId parameter."}
+	}
+	if req.Site != nil {
+		siteCopy := *req.Site
+		if siteCopy.Publisher == nil {
+			siteCopy.Publisher = &openrtb2.Publisher{ID: impExt.PublisherId}
+		} else {
+			publisher := *siteCopy.Publisher
+			publisher.ID = impExt.PublisherId
+			siteCopy.Publisher = &publisher
+		}
+		req.Site = &siteCopy
+
+	} else if req.App != nil {
+		appCopy := *req.App
+		if appCopy.Publisher == nil {
+			appCopy.Publisher = &openrtb2.Publisher{ID: impExt.PublisherId}
+		} else {
+			publisher := *appCopy.Publisher
+			publisher.ID = impExt.PublisherId
+			appCopy.Publisher = &publisher
+		}
+		appCopy.Publisher = &openrtb2.Publisher{ID: impExt.PublisherId}
+		req.App = &appCopy
+
+	}
+
+	return nil
+}
+
+func impressionByMediaType(imp *openrtb2.Imp) openrtb2.Imp {
+
+	impCopy := *imp
+	if imp.Banner != nil {
+		impCopy.Video = nil
+	}
+	if imp.Video != nil {
+		impCopy.Banner = nil
+
+	}
+	return impCopy
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb2.BidResponse
+
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(internalRequest.Imp))
+
+	// overrride default currency
+	if bidResp.Cur != "" {
+		bidResponse.Currency = bidResp.Cur
+	}
+
+	for _, sb := range bidResp.SeatBid {
+		for i := 0; i < len(sb.Bid); i++ {
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &sb.Bid[i],
+				BidType: getMediaTypeForImp(sb.Bid[i]),
+			})
+		}
+	}
+	return bidResponse, nil
+}
+
+// getMediaTypeForImp figures out which media type this bid is for.
+// SilverPush doesn't support multi-type impressions.
+// If both banner and video exist, take banner as we do not want in-banner video.
+func getMediaTypeForImp(bid openrtb2.Bid) openrtb_ext.BidType {
+	switch bid.MType {
+	case openrtb2.MarkupBanner:
+		return openrtb_ext.BidTypeBanner
+	case openrtb2.MarkupVideo:
+		return openrtb_ext.BidTypeVideo
+
```

Comment:
nitpick - please remove this blank line.

Reply:
Fixed
[commit](https://github.com/prebid/prebid-server/pull/2957/commits/41daf0668c923b8c72d21af84ad06a0f5fc47a62)